lib/mosaic-objects/src/main/java/org/eclipse/mosaic/lib/objects/addressing/AdHocMessageRoutingBuilder.java

```diff
@@ -64,109 +94,96 @@ private MessageRouting build(DestinationAddressContainer dac) {
-     */
-    public MessageRouting topoBroadCast() {
-        return topoCast(NetworkAddress.BROADCAST_ADDRESS.getAddress(), 1);
+    public AdHocMessageRoutingBuilder destination(String receiverName) {
+        return destination(IpResolver.getSingleton().nameToIp(receiverName));
     }
 
-    /**
-     * Creates a topological broadcast using a specific {@link AdHocChannel} and specific number of hops.
-     * Note: The SNS will dismiss hop value, since it only allows for single-hop TopoCasts
-     *
-     * @param hops number of hops
-     * @return MessageRouting
-     */
-    public MessageRouting topoBroadCast(int hops) {
-        return topoCast(NetworkAddress.BROADCAST_ADDRESS.getAddress(), hops);
+    public AdHocMessageRoutingBuilder destination(NetworkAddress ipAddress) {
+        assert !destinationSet: "Destination was already set! Using first setting.";
+        this.destination = ipAddress;
+        this.destinationSet = true;
+        return this;
     }
 
-    /**
-     * Creates a topological cast using a specific destination host name, a specific {@link AdHocChannel} and specific number of hops.
-     * Note: The SNS will dismiss hop value, since it only allows for single-hop TopoCasts, so if receiver can't be reached, within
-     * one hop transmission will fail.
-     *
-     * @param receiverName destination host name
-     * @param hops         number of hops
-     * @return MessageRouting
-     */
-    public MessageRouting topoCast(String receiverName, int hops) {
-        return topoCast(IpResolver.getSingleton().nameToIp(receiverName).getAddress(), hops);
+    public AdHocMessageRoutingBuilder broadcast() {
+        return destination(new NetworkAddress(NetworkAddress.BROADCAST_ADDRESS));
     }
 
-    /**
-     * Creates a topological cast using a specific destination IP address, a specific {@link AdHocChannel} and specific number of hops.
-     * Note: The SNS will dismiss hop value, since it only allows for single-hop TopoCasts, so if receiver can't be reached, within
-     * one hop transmission will fail.
-     *
-     * @param ipAddress destination IP address
-     * @param hops      number of hops
-     * @return MessageRouting
-     */
-    public MessageRouting topoCast(byte[] ipAddress, int hops) {
-        return build(new DestinationAddressContainer(
-                DestinationType.AD_HOC_TOPOCAST,
-                new NetworkAddress(ipAddress),
-                channel,
-                require8BitTtl(hops),
-                null,
-                ProtocolType.UDP
-        ));
+    public AdHocMessageRoutingBuilder topological() {
+        assert !routingSet: "Routing was already set! Using first setting";
+        this.routing = DestinationType.AD_HOC_TOPOCAST;
+        this.routingSet = true;
+        return this;
     }
 
-    /**
-     * The maximum time to live (TTL).
-     */
-    private final static int MAXIMUM_TTL = 255;
+    public AdHocMessageRoutingBuilder singlehop() {
```

Comment:
Do we want it to configure both? You could argue, that there is also a geographical singlehop message routing. We could simply remove the call to topological() and just return the Builder Object

Reply:
But then we have to discuss if what a call of `topological().broadcast().build()` results in, do we want to force the user to type `topological().broadcast().singlehop().build()` or `topological().broadcast().hops(1).build()` to make it work with SNS, OMNeT++ and NS-3, as they would not support just `topological().broadcast().build()`?

Reply:
I'm fine with both renaming to `topologicalSinglehop` or just configuring the hop and call `topological().singlehop()` 

From my personal taste I like the second option more, because it's more flexible, and not less readable.

Reply:
We agreed to do the second option.